tests/foreman/api/test_syncplan.py

```diff
@@ -1107,15 +1107,13 @@ def test_positive_synchronize_custom_product_weekly_recurrence(self):
             self.validate_task_status(repo.id, max_tries=2)
         self.validate_repo_content(
             repo, ['erratum', 'package', 'package_group'], after_sync=False)
-        # Wait the rest of expected time
+        # Update with the current UTC time plus a delay
+        sync_plan.sync_date = datetime.utcnow() + timedelta(seconds=delay)
```

Comment:
`sync_date` is time on which sync plan will start working. Your sync plan is set to do the job weekly, meaning it won't run until a week and `delay` from now :) .

You really need the previous version here:
```
start_date = datetime.utcnow() - timedelta(weeks=1) \
    + timedelta(seconds=delay)
```

Reply:
Ok, will fix, thanks for explaining.

I have since tested this "weekly sync plan with start date in the past" manually, on Sat6.5 and Sat6.4, and it worked.

Reply:
I also need to add `sync_plan.sync_date = start_date` if we go back to the previous version. Seems longer way to do it. Or did you mean this:
`sync_plan.sync_date = datetime.utcnow() - timedelta(weeks=1) \
    + timedelta(seconds=delay)`

Scanning other tests I see two of the former and five of the latter.

Reply:
Both are fine. You can assign calculation outcome to entity directly:
```
sync_plan.sync_date = datetime.utcnow() - timedelta(weeks=1) \
            + timedelta(seconds=delay)
```
or through variable:
```
start_date = datetime.utcnow() - timedelta(weeks=1) \
            + timedelta(seconds=delay)
sync_plan.sync_date = start_date
```

For me, preference for one over another is only matter of taste. Settling down on one convention in all cases sounds like good idea, but let's tackle it in separate PR.